### PR TITLE
fix-build

### LINKED ISSUE #8: fix failing build run

### PR DESCRIPTION
**Pull request by VortexExpansion**
Fixes https://github.com/diverta/front_next_media/issues/8

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -64,7 +64,6 @@ jobs:
   #for Push
   pushed_build:
     name: Build(pushed)
-    if: (github.repository == 'diverta/front_next_media' && (github.event_name == 'push' || github.event_name == 'workflow_dispatch'))
```
Comment (yabe-diverta) [inline .github/workflows/build.yml:67]: Why did you remove this ?

**Comment by VortexExpansion** [comment]
@yabe-diverta @HisashiHasebe san 
Ready for review

File: src/app/(pages)/(TwoColumn)/article/[id]/page.jsx [hunk 2]
```diff
@@ -6,6 +6,7 @@ import TagArea from '@/components/common/TagArea'
 import TagKeyword from '@/components/common/TagKeyword'
 import Feature from '@/components/section/feature/Feature'
 
+export const dynamic = 'force-static';
```
Comment (yabe-diverta) [inline src/app/(pages)/(TwoColumn)/article/[id]/page.jsx]: @VortexExpansion 
no needed?
Because we defines this website supposed to be generated by server-side (SSG).
Please tell me what cases is this option needed?

File: src/app/(pages)/(TwoColumn)/article/page.jsx [hunk 3]
```diff
@@ -9,6 +9,7 @@ import PageTitle from '@/components/common/PageTitle'
 import TagArea from '@/components/common/TagArea'
 import TagKeyword from '@/components/common/TagKeyword'
 
+export const dynamic = 'force-static';
```
Comment (yabe-diverta) [inline src/app/(pages)/(TwoColumn)/article/page.jsx]: same above

**Review by yabe-diverta — COMMENTED** [review]
Review finished, please check comments 🙇

File: src/app/(pages)/(TwoColumn)/article/page.jsx (hunk 3 above)
Comment (VortexExpansion) [reply src/app/(pages)/(TwoColumn)/article/page.jsx]: @yabe-diverta san

This is what was making the `build` fail. 
Because the searchQueryParams make this page dynamic. 
So I had to force it to be static for fixing the build make. 

Build fail error : https://diverta.gyazo.com/ae6b90f0c2fb98c74078dcfdeecf61d2
Reference for solving : https://nextjs.org/docs/messages/app-static-to-dynamic-error

File: src/app/(pages)/(TwoColumn)/article/[id]/page.jsx (hunk 2 above)
Comment (VortexExpansion) [reply src/app/(pages)/(TwoColumn)/article/[id]/page.jsx]: Checked build again. This is redundant. 
Removed.

File: src/app/(pages)/(TwoColumn)/article/page.jsx (hunk 3 above)
Comment (yabe-diverta) [reply src/app/(pages)/(TwoColumn)/article/page.jsx]: I believe this page should work on some sonditions like `page` or `tag_id`,
Although those conditions are not able to get from server-side.
Apparently using `force-dynamic` is injecting null into `searchParams` param ([ref](https://github.com/vercel/next.js/issues/50999#issuecomment-1727941641)).
So my suggestion is how about using client-side fetch for this page?

**Comment by yabe-diverta** [comment]
@VortexExpansion 
Sry adding issue more.
To verify build output I had run `npm run build` on my local, then found 2 issues.
- outputs error log
```
TypeError: Cannot read properties of undefined (reading 'contents_type_nm')
    at h (/Users/yabe/workspace/front_next_media/.next/server/app/(pages)/(TwoColumn)/article/[id]/page.js:1:4467)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Error occurred prerendering page "/article/[id]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'contents_type_nm')
    at h (/Users/yabe/workspace/front_next_media/.next/server/app/(pages)/(TwoColumn)/article/[id]/page.js:1:4467)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
TypeError: Cannot read properties of undefined (reading 'subject')
    at l (/Users/yabe/workspace/front_next_media/.next/server/app/(pages)/(TwoColumn)/feature/[id]/page.js:1:3057)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:35049)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:47638)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51060)
    at nm (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:54300)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:45143)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51060)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51249)
    at nm (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:54300)
    at nh (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:53366)

Error occurred prerendering page "/feature/[id]". Read more: https://nextjs.org/docs/messages/prerender-error
TypeError: Cannot read properties of undefined (reading 'subject')
    at l (/Users/yabe/workspace/front_next_media/.next/server/app/(pages)/(TwoColumn)/feature/[id]/page.js:1:3057)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:35049)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:47638)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51060)
    at nm (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:54300)
    at nf (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:45143)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51060)
    at np (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:51249)
    at nm (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:54300)
    at nh (/Users/yabe/workspace/front_next_media/node_modules/next/dist/compiled/next-server/app-page.runtime.prod.js:45:53366)

> Export encountered errors on following paths:
        /(pages)/(TwoColumn)/article/[id]/page: /article/[id]
        /(pages)/(TwoColumn)/feature/[id]/page: /feature/[id]
```
- found no `index.html` and the other HTMLs exported
```sh
~/w/front_next_media ❯❯❯ tree out -L 1
out
├── 404.html
├── _next
├── css
├── favicon.ico
├── images
├── js
├── kuroco_front.json
├── login.html
├── login.txt
├── mock
├── next.svg
├── privacy.html
├── privacy.txt
├── svg
└── vercel.svg

6 directories, 9 files
```

In fact I checked actual behavior with running `npx serve out` then top page was missing.
Could you check?

File: src/app/(pages)/(TwoColumn)/article/page.jsx (hunk 3 above)
Comment (VortexExpansion) [reply src/app/(pages)/(TwoColumn)/article/page.jsx]: I might be wrong but I think that won't work too. 
Because if you look at [reminder/page.jsx](https://github.com/diverta/front_next_media/blob/main/src/app/(pages)/(SigleColumn)/reminder/page.jsx#L11), this is an example of using `searchParams` with `use-client` 

But this page still seems to fail at `npm run build`. with or without the `force-static` constraint

Please check once

**Comment by takel-shiba** [comment]
@yabe-diverta 
サポートお願いします

**Comment by HisashiHasebe** [comment]
@yabe-diverta @VortexExpansion 
I fixed build error.
Can you check it?

```
> next-media@0.1.0 build
> next build

   ▲ Next.js 14.0.4
   - Environments: .env.local, .env

 ✓ Creating an optimized production build
 ✓ Compiled successfully
 ✓ Linting and checking validity of types
 ✓ Collecting page data
 ✓ Generating static pages (49/49)
 ✓ Finalizing page optimization

Route (app)                              Size     First Load JS
┌ λ /                                    190 B          93.8 kB
├ ○ /_not-found                          875 B          82.8 kB
├ λ /article                             977 B          94.6 kB
├ ● /article/[id]                        2.99 kB        96.6 kB
├   ├ /article/39
├   ├ /article/22
├   ├ /article/16
├   └ [+31 more paths]
├ ● /feature/[id]                        1.28 kB        94.9 kB
├   ├ /feature/20
├   └ /feature/17
├ ○ /login                               3.15 kB        91.9 kB
├ ○ /member/delete                       1.32 kB        92.7 kB
├ ○ /member/edit                         1.68 kB        93.1 kB
├ ○ /member/mypage                       1.86 kB        98.2 kB
├ ○ /privacy                             137 B          82.1 kB
├ ○ /register                            3.4 kB         92.1 kB
└ ○ /reminder                            2.72 kB        91.4 kB
+ First Load JS shared by all            82 kB
  ├ chunks/938-1e956c3b7103a91a.js       26.8 kB
  ├ chunks/fd9d1056-3105055c309dacd0.js  53.3 kB
  ├ chunks/main-app-8e68421c24d9d3ad.js  220 B
  └ chunks/webpack-07b3f12c022c8618.js   1.68 kB


○  (Static)   prerendered as static content
●  (SSG)      prerendered as static HTML (uses getStaticProps)
λ  (Dynamic)  server-rendered on demand using Node.js
```

**Comment by VortexExpansion** [comment]
Thank you for your contribution 🙇  : @HisashiHasebe san
This is great actually. The page reminder error has been fixed. :sparkles: 

I tried to run the workflow and generate the static pages again locally 
https://github.com/VortexExpansion/front_next_media/actions/runs/8152774447/job/22282852282
You can see that all the pages are loaded but fails at the end for `undefined contents_type_nm`  for `article/[id]`
https://diverta.gyazo.com/254c7ca4ddb9fa2db4d8dbb69baf32c0
And this behaviour is only when you run the workflow. On running `npm run build` locally, you don't get this error

So the searchQueryParams in `article/page.jsx` or `article/[id]/page.jsx` still needs to be fixed

@yabe-diverta san please check 🙇

**Comment by yabe-diverta** [comment]
@VortexExpansion 
~~ref to [this comment](https://github.com/diverta/front_next_media/pull/18#issuecomment-1985026813),~~
~~We would change cache strategy, so maybe no need to consider the searchQueryParams issue.~~
We should not remove current behavior regarding fetch, still need to consider SSR.

File: src/app/(pages)/(TwoColumn)/article/page.jsx (hunk 3 above)
Comment (yabe-diverta) [reply src/app/(pages)/(TwoColumn)/article/page.jsx]: This page is successfully generated by build command.
So this conversation is a bit stale.
Closing.

**Comment by yabe-diverta** [comment]
@VortexExpansion 
fixed many overall, including other topics 🙇
maybe it's hard to review because of too many changes, 
please check it's right to work on browser.

including interests:
- can build
- can route to limited page dynamically
- simplify app/pages
- check not to loading/fetching data in infinite loop
- linter/formatter
- providing custom not-found/error screen
- gather fetcher fns to fetch dir

**Comment by VortexExpansion** [comment]
@yabe-diverta san
Problems I noticed, 
I will continue to fix them, you may also if you have time

- [x] Broken favourite icon for myFavourites https://diverta.gyazo.com/fcf9610fd546b96ff5da0fe8569caa6b
- [x] Broken favourite icon for feature articles https://diverta.gyazo.com/a4934d0b7738e3ddb95a0068f50df379
- [x] Wrong redirect on registration https://diverta.gyazo.com/cf26e5ff70d38ea01c583e6744de0f8f
- [x] Broken pager https://diverta.gyazo.com/1dff3f2c5200da1ae5ccd7d800634ae1
- [x] Cannot search by tags https://diverta.gyazo.com/9b55fe89256598433eae358ccc5cc8e8 
- [x] delete favourite button functionality broken https://diverta.gyazo.com/2c31d1eb7e472e2c45b4c0cde6edcd53

**Comment by VortexExpansion** [comment]
Build works well including the dynamic pages ✨ 

@HisashiHasebe san please check if you find any problems

**Comment by HisashiHasebe** [comment]
@VortexExpansion 
`npm run build` is work well but, I try to check with `npm run dev` and I got error.
Can you check it?

https://diverta.gyazo.com/e6fe49ef42a7622bd970b0b375045be9
https://diverta.gyazo.com/eee77032051342bf494d8efc1b0f000d

**Comment by VortexExpansion** [comment]
@HisashiHasebe san
I believe that is just next cache problem on your local.
Please try ->

- Delete `.next` `node_modules` and `package-lock.json` from root folder
- Run `npm install`
- Run `npm run dev`

Should work fine.

**Comment by HisashiHasebe** [comment]
@VortexExpansion 
Thank you! It's work fine.
Please merge this PR.

**Comment by VortexExpansion** [comment]
@yabe-diverta san please merge.

I still don't have write access to this repository 🙇
